woningwaardering/stelsels/utils.py

```diff
@@ -236,46 +236,81 @@ def energieprestatie_met_geldig_label(
         eenheid (EenhedenEenheid): De eenheid met mogelijke energieprestaties.
 
     Returns:
-        EenhedenEnergieprestatie | None: De eerste geldige energieprestatie met een energielabel en None Wanneer er geen geldige energieprestatie met label is gevonden.
+        EenhedenEnergieprestatie | None: De eerst geldige energieprestatie en None wanneer er geen geldige energieprestatie met label is gevonden.
     """
-    if eenheid.energieprestaties is not None:
-        for energieprestatie in eenheid.energieprestaties:
-            if (
-                energieprestatie.registratiedatum
-                and energieprestatie.soort
-                and energieprestatie.soort.code
-                and energieprestatie.status
-                and energieprestatie.status.code
-                and energieprestatie.begindatum
-                and energieprestatie.einddatum
-                and energieprestatie.label
-                and (
-                    energieprestatie.soort.code
-                    in [
-                        Energieprestatiesoort.energie_index.code,
-                        Energieprestatiesoort.primair_energieverbruik_woningbouw.code,
-                        Energieprestatiesoort.voorlopig_energielabel.code,  # Een voorlopig energie_label kan ook als status definitief zijn, want dit is het soort energie label gemeten met de meetmethode van voor 2015.
-                    ]
-                )
-                and (
-                    energieprestatie.begindatum < peildatum < energieprestatie.einddatum
-                    and energieprestatie.status.code
-                    == Energieprestatiestatus.definitief.code
-                )
-                and (
-                    # Check of de registratie niet ouder is dan 10 jaar
-                    energieprestatie.registratiedatum
-                    > (
-                        datetime.combine(peildatum, time.min).astimezone()
-                        - relativedelta(years=10)
-                    )
-                )
-            ):
-                logger.info("Energieprestatie met geldig label gevonden")
-                logger.debug(f"Energieprestatie: {energieprestatie}")
-                return energieprestatie
 
-    logger.info("Geen geldige energieprestatie met label gevonden")
+    if eenheid.energieprestaties is None:
+        warnings.warn(f"Eenheid {eenheid.id}: 'Energieprestaties' is None", UserWarning)
+        return None
+
+    vereiste_attributen = [
+        ("registratiedatum", lambda ep: ep.registratiedatum),
+        ("soort", lambda ep: ep.soort and ep.soort.code),
+        ("status", lambda ep: ep.status and ep.status.code),
+        ("begindatum", lambda ep: ep.begindatum),
+        ("einddatum", lambda ep: ep.einddatum),
+        ("label", lambda ep: ep.label and ep.label.code),
+    ]
```

Comment:
Je kan dit typed maken op deze manier, dan kan de `# type: ignore [no-untyped-call]` op regel 260 ook weg;
```suggestion
    vereiste_attributen: List[Tuple[str, Callable[[EenhedenEnergieprestatie], bool]]] = [
        ("registratiedatum", lambda ep: ep.registratiedatum is not None),
        ("soort", lambda ep: ep.soort is not None and ep.soort.code is not None),
        ("status", lambda ep: ep.status is not None and ep.status.code is not None),
        ("begindatum", lambda ep: ep.begindatum is not None),
        ("einddatum", lambda ep: ep.einddatum is not None),
        ("label", lambda ep: ep.label is not None and ep.label.code is not None),
    ]
```